.github/pull_request_template.md

```diff
@@ -0,0 +1,33 @@
+# Problem Statement
+- Problem statement
+
+# Design
+-  For Bug, Describe the fix here.
+-  For Feature, Post the link for design
+
+# Coding
+   Checklist for Author
+-  [ ] Coding conventions are followed and code is consistent
+
+# Testing 
+  Checklist for Author
+- [ ] Unit and System Tests are added
+- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
+- [ ] Testing was performed with RPM
+
+# Impact Analysis
+  Checklist for Author/Reviewer/GateKeeper
+- [ ] Interface change (if any) are documented
+- [ ] Side effects on other features (deployment/upgrade)
+- [ ] Dependencies on other component(s)
+
+# Review Checklist 
+  Checklist for Author
+- [ ] JIRA number/GitHub Issue added to PR
+- [ ] PR is self reviewed
+- [ ] Jira and state/status is updated and JIRA is updated with PR link
```

Comment:
`s/Jira/JIRA/`
Let's keep the naming consistent within a single document.

Reply:
The word update doesn't tell what exactly is expected to happen to JIRA ticket. Perhaps a link to the document is required telling how PR workflow maps to JIRA ticket workflow.